onyo/lib/onyo.py

```diff
@@ -85,12 +85,19 @@ def templates(self) -> set[Path]:
 
         return self._templates
 
-    def clear_caches(self) -> None:
-        self._assets = None
-        self._dirs = None
-        self._files = None
-        self._gitfiles = None
-        self._templates = None
+    def clear_caches(self, assets: bool = True, dirs: bool = True,
```

Comment:
Just so I understand the intended use pattern: calling this function will either clear everything, or an explicit list of what /not/ to clear needs to be specified.

The listing of negatives feels backwards to me. Perhaps a list of caches, that is fully populated by default.

Reply:
This is probably the main thing I was thinking about quite a lot since yesterday, and I can't say that you are wrong.

I personally feel like having booleans and being forced to explicetely state deactivation is from API usage perspective the most comfortable way to use it. I don't want to check everytime all caches the Repo class offers, just to build a list of the 4-5 of the six I want to activate. Either I want them all to be cleared, because something is changed, or I should be sure that this specific cache is not needed.

We can discuss this more deeply in a meeting or in here if you want, though my current thoughts all go back to "this is very different from the fsck functionality" and the usage like this seems most comfortable and intuitive to me.